### ISSUE TITLE
Make promises part of documentation more clear

### ISSUE THREAD

**Issue by elhigu** [knex/documentation#243]
Currently documentation basically says that calling `.then` makes query builder to a promise, which is not true.

http://knexjs.org/#Interfaces-Promises

Some discussion about it is found in https://github.com/knex/knex/issues/3515#issuecomment-550201065
but the whole section should be revised.

**Comment by TotalTechGeek** [comment]
I agree with this, I also believe that the documentation needs to provide a caveat emptor on using async / await with knex queries, as they will not begin execution unless you call `.then` or `await` on the query.

Thus code you might write expect is running in parallel actually is running each of the queries one at a time, and could lead a novice programmer to misunderstand how promises are supposed to function in JS (as it did within my team). 

```js
 let result1 = knex('login_users').select('email').innerJoin('users_friends', 'users_friends.user2', 'login_users.id').where({
     user1: user,
     accepted: true
 })

 let result2 = knex('login_users').select('email').innerJoin('users_friends', 'users_friends.user1', 'login_users.id').where({
     user2: user,
     accepted: false
 })

 let result3 = knex('login_users').select('email').innerJoin('users_friends', 'users_friends.user2', 'login_users.id').where({
     user1: user,
     accepted: false
 })

 res.json([{
     Friends: await result1
 }, {
     FriendRequestsToYou: await result2
 }, {
     FriendRequestsFromYou: await result3
 }])
```

This code works, but it doesn't execute the queries in parallel (it's not meant to, but that's not necessarily clear to someone new to the framework). 

We had an incident where a developer wrote fully functioning code using Async / Await, but it didn't feel quite right

```js
const updateFriendsQuery = knex('users_friends')
.where({user1: friendId, user2: yourId})
.update({accepted: true})

if(someOtherQuerySuccess)
{
    // executes updateFriendsQuery
    await updateFriendsQuery
}
```

This worked as the newer programmer intended, but confused some of the more senior developers. He was under the impression that promises would not begin to execute unless you explicitly called `.then` or performed an `await` on them. 

If there were some sort of caveat emptor on knex, saying "You need to add `.then()` to get the query to run as a promise", or some other method alias were added, I believe it would lead to code that better fit the async / await paradigm. 

--- 

To the Knex.js developers, do not take this as a criticism of your work, I **fully** understand why the QueryBuilder chain can't immediately return a promise.

I just believe it should be more explicitly called out, so I might try to file a PR.
